components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -80,17 +80,60 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
 
         val geckoResult = GeckoResult<PromptResponse>()
         val onConfirmSave: (Login) -> Unit = { login ->
-            geckoResult.complete(prompt.confirm(login.toLoginEntry()))
+            geckoResult.complete(prompt.confirm(Autocomplete.LoginSelectOption(login.toLoginEntry())))
         }
         val onDismiss: () -> Unit = {
             geckoResult.complete(prompt.dismiss())
         }
 
         geckoEngineSession.notifyObservers {
             onPromptRequest(
-                PromptRequest.LoginPrompt(
-                    hint = prompt.hint,
-                    logins = prompt.logins.map { it.toLogin() },
+                PromptRequest.SaveLoginPrompt(
+                    hint = prompt.options[0].hint,
+                    logins = prompt.options.map { it.value.toLogin() },
+                    onConfirm = onConfirmSave,
+                    onDismiss = onDismiss
+                )
+            )
+        }
+        return geckoResult
+    }
+
+    override fun onLoginSelect(
+        session: GeckoSession,
+        prompt: PromptDelegate.AutocompleteRequest<Autocomplete.LoginSelectOption>
+    ): GeckoResult<PromptResponse>? {
+        fun Autocomplete.LoginEntry.toLogin() = Login(
```

Comment:
This isn't actually doing anything, now we ignore this prompt type for now. Filed https://github.com/mozilla-mobile/android-components/issues/7134 about following up for the feature

Reply:
We can document it in the code and link to the issue :)